### PR TITLE
User Profiles

### PR DESCRIPTION
**Pull request by mattss**
Initial implementation of user profiles, including:
- Content-based user accounts (using dexterity.membrane)
- Behaviour-based profile information to aid customisation
- Testing content in ploneintranet.suite now now creates proper user profiles
- Restored personal bar in top-right
- Basic user profile view including:
  - Basic user information
  - Follow button (almost works)
  - Personal timeline

### CODE REVIEW

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/116/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/117/
Test FAILed.

**Comment by mattss** [comment]
Wasn't seeing those failures locally. Will investigate.

**Comment by mattss** [comment]
Retest this please Jenkins

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/118/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/119/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/120/
Test FAILed.

**Comment by mattss** [comment]
As far as I can tell the only remaining issue here is a known heisenbug related to the 'Close' buttons - see https://github.com/ploneintranet/ploneintranet/blob/master/src/ploneintranet/suite/tests/acceptance/workspace.robot#L385

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/126/
Test PASSed.

**Comment by mattss** [comment]
Tests passing! I shall update this and fix merge issues

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/127/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/135/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/136/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/137/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/139/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/140/
Test PASSed.

**Comment by mattss** [comment]
Build passing! I'm going to have a call with @cillianderoiste later regarding profiles, so I suggest we hold off merging this until then.

File: src/ploneintranet/userprofile/browser/configure.zcml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    i18n_domain="ploneintranet">
+
+  <browser:page
+      name="userprofile-view"
+      for="ploneintranet.userprofile.content.userprofile.IUserProfile"
+      class=".userprofile.UserProfileView"
+      permission="zope2.View"
+      template="templates/userprofile.pt"
+      />
+
```
Comment (gyst) [inline src/ploneintranet/userprofile/browser/configure.zcml:18]: Missing binding for userprofile layer

File: src/ploneintranet/userprofile/browser/configure.zcml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <browser:page
+      for="Products.CMFPlone.interfaces.IPloneSiteRoot"
+      name="author"
+      class=".userprofile.AuthorView"
+      permission="zope2.View"
+      layer="ploneintranet.theme.interfaces.IThemeSpecific"
+     />
```
Comment (gyst) [inline src/ploneintranet/userprofile/browser/configure.zcml:25]: That is incorrect. You're not allowed to bind to the theme layer here. You should bind to a userprofile layer.
We want to be able to re-use this package with another theme.

File: src/ploneintranet/userprofile/browser/configure.zcml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  <browser:page
+      for="Products.CMFPlone.interfaces.IPloneSiteRoot"
+      name="useractions"
+      class=".userprofile.UserActions"
+      permission="zope2.View"
+      layer="ploneintranet.theme.interfaces.IThemeSpecific"
+     />
```
Comment (gyst) [inline src/ploneintranet/userprofile/browser/configure.zcml:33]: idem

File: src/ploneintranet/userprofile/profiles/default/types/ploneintranet.userprofile.userprofile.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ <property name="immediate_view">userprofile-view</property>
+ <property name="global_allow">False</property>
+ <property name="allow_discussion">False</property>
+ <property name="default_view">userprofile-view</property>
+ <property name="view_methods">
+  <element value="userprofile-view"/>
+ </property>
+ <property name="default_view_fallback">False</property>
+ <property name="add_permission">cmf.AddPortalContent</property>
+ <!-- schema and class used for content items -->
```
Comment (gyst) [inline src/ploneintranet/userprofile/profiles/default/types/ploneintranet.userprofile.userprofile.xml:18]: WOW! That should be cmf.AddPortalMember!!!

File: src/ploneintranet/userprofile/profiles/default/types/ploneintranet.userprofile.userprofilecontainer.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  <!-- View information -->
+  <property name="link_target"></property>
+  <property name="immediate_view">view</property>
+  <property name="default_view">view</property>
+  <property name="view_methods">
+   <element value="view"/>
+  </property>
+  <property name="default_view_fallback">False</property>
+  <property name="add_permission">cmf.AddPortalContent</property>
+
```
Comment (gyst) [inline src/ploneintranet/userprofile/profiles/default/types/ploneintranet.userprofile.userprofilecontainer.xml:37]: This should be ManagePortal

File: src/ploneintranet/userprofile/profiles/default/workflows/userprofile_workflow/definition.xml [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+  <transition transition_id="approve" new_state="enabled"
+               title="Approve this member"
+               trigger="USER"
+               before_script="" after_script="">
+   <action url="%(content_url)s/content_status_modify?workflow_action=approve"
+           category="workflow">Approve This Member</action>
+   <guard>
+    <guard-permission>Review portal content</guard-permission>
+   </guard>
```
Comment (gyst) [inline src/ploneintranet/userprofile/profiles/default/workflows/userprofile_workflow/definition.xml:90]: Doesn't this take treating members as content too far? I know this is the canonical membrane way, but the security implications of publishing a page are different from those of enabling a user account. Would like some discussions on this.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/157/
Test FAILed.

**Comment by mattss** [comment]
Retest this please jenkins

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/160/
Test FAILed.

File: src/ploneintranet/network/browser/follows.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+            self.verb = _(u'Follow')
+
+        self.unique_id = uuid.uuid4().hex
+
+        return self.index()
+
+    def toggle_follow(self):
+        """Follow/unfollow context."""
+        if self.request.get('REQUEST_METHOD') != 'POST':
+            raise NotAllowed("Write on POST only.")
```
Comment (pbauer) [inline src/ploneintranet/network/browser/follows.py]: plone.protect.postonly.check is the convenience-method for that

**Comment by pbauer** [comment]
I reviewed the pr and it is good to merge once the tests are fixed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/176/
Test PASSed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/177/
Test FAILed.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/178/
Test PASSed.

**Pull request merged**